src/commands/gcp-oidc-authorize.yml

```diff
@@ -0,0 +1,86 @@
+description: Authorize the GCP CLI using OIDC rather than receiving a service account's credentials via environment variable. This command creates an ephemeral GCP session by leveraging GCP Workload Identity. Workload Identity must be configured in the project you are attempting to authenticate to prior to using this command. Additionally, you will need to have installed the gcloud command line utility in your CI execution environment prior to using this command. This is a enhanced version of the gcp-cli orb's authorize command.
+parameters:
+  do-docker-login:
+    type: enum
+    default: ""
+    enum: ["", "true"]
+    description: "Have Docker use the retrieved creds to log into Google Artifact Registry, where we have a Docker registry. Likely will need to run setup_remote_docker command first. "
+  do-helm-login:
+    type: enum
+    default: ""
+    enum: ["", "true"]
+    description: "Have Helm use the retrieved creds to log into Google Artifact Registry, using Helm's support for OCI registries."
+  do-activate-application-default-creds:
+    type: enum
+    default: "true"
+    enum: ["","true"]
+    description: "Configure the environment to use the retrieved credentials as the Application Default Creds (ADC). This option does not work if delete-cred-file is true"
+  delete-cred-file:
+    type: enum
+    default: "false"
+    enum: ["false", "true"]
+    description: delete the credentials file
+  project-id:
+    type: env_var_name
+    default: GCP_PROJECT_ID
+    description: The name of the environment variable containing the GCP project ID to be authenticated to. For example, the named variable could contain "myproject-123456"
+  workload-identity-pool-id:
+    type: env_var_name
+    default: GCP_WIP_ID
+    description: The name of the environment variable containing the workload identity pool to use to authenticate to GCP
+  workload-identity-pool-provider-id:
+    type: env_var_name
+    default: GCP_WIP_PROVIDER_ID
```

Comment:
ohhhh, and users of this command probably should create contexts with these variables defined within them.... I mean, they don't have to of course, but that would be the easiest way

Reply:
Yep 100% agreed and that's the route we plan to go.

Just to share, internally, the [Terraform module](https://github.com/mdg-private/platform-terraform-modules/tree/db1cc0c98f1892a8df6edb88909ebd430021c522/modules/infrastructure/cci_federation) that creates the necessary Workload Identity Federation config can also optionally create the necessary contexts.